### PR TITLE
feat(general): number of words larger/less than or equal operators

### PR DESCRIPTION
**Pull request by kadoshms**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

Added support for four new operators:

`number_of_words_greater_than`
`number_of_words_greater_than_or_equal`
`number_of_words_less_than`
`number_of_words_less_than_or_equal`

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/checks_infra/solvers/attribute_solvers/base_number_of_words_attribute_solver.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from typing import Optional, Any, Dict
+
+from checkov.common.graph.checks_infra.enums import Operators
+from checkov.common.checks_infra.solvers.attribute_solvers.base_attribute_solver import BaseAttributeSolver
+from checkov.common.util.type_forcers import force_int
+
+
+class BaseNumberOfWordsAttributeSolver(BaseAttributeSolver):
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/base_number_of_words_attribute_solver.py:13]: fancy, creating a base class 🏆

File: tests/terraform/graph/checks_infra/attribute_solvers/number_of_words_greater_than_or_equal_solver/test_solver.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import os
+
+from tests.terraform.graph.checks_infra.test_base import TestBaseSolver
+
+TEST_DIRNAME = os.path.dirname(os.path.realpath(__file__))
+
+
+class TestNumberOfNotWordsGreaterThanOrEqual(TestBaseSolver):
```
Comment (gruebel) [inline tests/terraform/graph/checks_infra/attribute_solvers/number_of_words_greater_than_or_equal_solver/test_solver.py:8]: all the test classes have a `Not` inside on purpose or copy-paste? 😄

**Review by gruebel — COMMENTED** [review]
nice job, I wanted to ask in the other PR, if you also will implement greater and less flavours 😄  I also fixed linting and typing for you, expect you looked forward on doing it on you own 😉

File: tests/terraform/graph/checks_infra/attribute_solvers/number_of_words_greater_than_or_equal_solver/test_solver.py (hunk 2 above)
Comment (kadoshms) [reply tests/terraform/graph/checks_infra/attribute_solvers/number_of_words_greater_than_or_equal_solver/test_solver.py:8]: yes (haha) will fix :)

**Review by gruebel — APPROVED** [review]
noice 🍻

**Review by rotemavni — APPROVED** [review]

File: checkov/common/checks_infra/solvers/attribute_solvers/number_of_words_greater_than_or_equal_attribute_solver.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+from typing import Optional, Any, Dict
+
+from checkov.common.checks_infra.solvers.attribute_solvers.base_number_of_words_attribute_solver import \
+    BaseNumberOfWordsAttributeSolver
+from checkov.common.graph.checks_infra.enums import Operators
+
+
+class NumberOfWordsGreaterThanOrEqualAttributeSolver(BaseNumberOfWordsAttributeSolver):
+    operator = Operators.NUMBER_OF_WORDS_GREATER_THAN  # noqa: CCE003  # a static attribute
```
Comment (YaaraVerner) [inline checkov/common/checks_infra/solvers/attribute_solvers/number_of_words_greater_than_or_equal_attribute_solver.py:9]: I think it's the wrong operator 🤔

File: checkov/common/checks_infra/solvers/attribute_solvers/number_of_words_less_than_or_equal_attribute_solver.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from typing import Optional, Any, Dict
+
+from checkov.common.checks_infra.solvers.attribute_solvers.base_number_of_words_attribute_solver import \
+    BaseNumberOfWordsAttributeSolver
+from checkov.common.graph.checks_infra.enums import Operators
+
+
+class NumberOfWordsLessThanOrEqualAttributeSolver(BaseNumberOfWordsAttributeSolver):
+    operator = Operators.NUMBER_OF_WORDS_GREATER_THAN  # noqa: CCE003  # a static attribute
```
Comment (YaaraVerner) [inline checkov/common/checks_infra/solvers/attribute_solvers/number_of_words_less_than_or_equal_attribute_solver.py:9]: also here

**Pull request merged**